src/parse_instruction.py

```diff
@@ -143,6 +156,10 @@ class Instruction:
     args: List[Argument]
     meta: InstructionMeta
 
+    inputs: List[Access]
+    outputs: List[Access]
+    clobbers: List[Access]
```

Comment:
I added a comment to try to clarify overlapping, but you're right it's potentially confusing.

I'm worried that keeping outputs/clobbers disjoint will be a lot of work without much benefit? e.g. in ` narrow_func_call_outputs`, it would have to move the accesses from outputs to clobbers?